### PR TITLE
Added branding exception to the ProviderPickerActivity

### PR DESCRIPTION
**Pull request by dxslly**
Google's provider implementation lives inside the Google Play Services APK which has a different application label and icon than user's expect. To avoid confusion, a special case is made to provide the correct branding. Ideally this should be solved at the protocol level (e.g. optional metadata specified in the manifest that can override these defaults for trusted providers).

### CODE REVIEW

File: api/java/org/openyolo/api/internal/ProviderPickerActivity.java [hunk 1]
```diff
@@ -234,8 +235,24 @@ public static Intent createDeleteIntent(
...
                     ApplicationInfo info = pm.getApplicationInfo(packageName, 0);
-                    mProviderIcons.add(info.loadIcon(pm));
-                    mProviderNames.add(pm.getApplicationLabel(info).toString());
+
+                    // NOTE(dxslly): Google's provider implementation lives inside the Google Play
+                    // Services APK which has a different application label and icon than user's
+                    // expect. To avoid confusion, a special case is made to provide the correct
+                    // branding. Ideally this should be solved at the protocol level (e.g. optional
+                    // metadata specified in the manifest that can override these defaults for
+                    // trusted providers).
```
Comment (StanKocken) [inline api/java/org/openyolo/api/internal/ProviderPickerActivity.java]: I agree that we should be able to define this from the provider manifest.
I don't really like the fact that we introduce provider-specific images and text within this repository. 
I understand that you need this right now and I'm ok to merge it, but we should plan to replace it as soon as possible.

Comment (nerdyverde) [reply api/java/org/openyolo/api/internal/ProviderPickerActivity.java]: I'm also okay with this as a temporary measure, but agree that this should be handled properly in subsequent releases.

Comment (dxslly) [reply api/java/org/openyolo/api/internal/ProviderPickerActivity.java]: Agreed, this should be only be allowed temporarily for the Droidcon release. I have filed https://github.com/openid/OpenYOLO-Android/issues/96 to track it's removal.

Comment (dxslly) [reply api/java/org/openyolo/api/internal/ProviderPickerActivity.java]: If everyone is okay with this temporary fix, would you mind marking the PR as approved?  Thanks!

**Pull request merged**